pkg/autodiscover/autodiscover.go

```diff
@@ -280,3 +290,43 @@ func getOpenshiftVersion(oClient clientconfigv1.ConfigV1Interface) (ver string,
 
 	return "", errors.New("could not get openshift version from clusterOperator")
 }
+
+// Get the list of pods mananged by the operator - retrieve from the CSV and then check the operator group to find the target namespaces to retrieve the managed pods
+func getOperatorCsvPods(csv *olmv1Alpha.ClusterServiceVersion) []corev1.Pod {
+	annotations := csv.Annotations
+
+	targetNamespaces := annotations["olm.targetNamespaces"]          // This is a comma-separated string, example : a,b,c where a, b and c are target namespaces
```

Comment:
The targetNamespaces is an attribute of the operatorGroup, which can be shared by multiple operators. So if 2 operators op1 and op2 share the same operator group, and only op1 is selected via label, this code is going to catch pods from op1 and op2, whereas we should only catch pods from op1.

Reply:
True, then we do not have a way to get only op1 related pods. Because there is no mapping between these pods created by the controller of the operator op1.

Reply:
Thanks @edcdavid 👍  I updated the code as discussed.